src/core/StringX.h

```diff
@@ -28,6 +28,8 @@
 #include "../os/typedefs.h"
 #include "./PwsPlatform.h"
 
+extern void trashMemory(void *buffer, size_t length);
```

Comment:
It is because Util.h contains references to StringX in it's header file. The StringX class needs Util.h for trashMemory(), but Util.h needs StringX to be complete for the references in Util.h to resolve, hence the circular dependancy. It's hard to describe in words, it might be easier to try it and see for yourself. I've updated the comments for clarification.